core/arch/x86/mangle.c

```diff
@@ -4120,7 +4126,8 @@ finalize_selfmod_sandbox(dcontext_t *dcontext, fragment_t *f)
     if (FRAGMENT_SELFMOD_COPY_CODE_SIZE(f) > 1) {
         pc = FCACHE_ENTRY_PC(f) + selfmod_copy_end_offs[i][j]IF_X64([k]);
         /* subtract the size itself, stored at the end of the copy */
-        *((cache_pc*)pc) = (copy_pc + FRAGMENT_SELFMOD_COPY_CODE_SIZE(f));
+        /* i#2155: -1 is to start comparison in the copy. */
```

Comment:
Please elaborate: something about "the starting point for the *backward* repe cmps".

Could merge w/ comment above.